docker/install-llvm-mlir.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+apt remove -y --purge --auto-remove cmake
+apt update && \
+   apt install -y software-properties-common lsb-release && \
+   apt clean all
+wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | tee /etc/apt/trusted.gpg.d/kitware.gpg >/dev/null
+apt-add-repository "deb https://apt.kitware.com/ubuntu/ $(lsb_release -cs) main"
+apt update
+apt install -y kitware-archive-keyring
+rm /etc/apt/trusted.gpg.d/kitware.gpg
+apt update
+apt install -y cmake
+
+wget https://github.com/llvm/llvm-project/releases/download/llvmorg-${VERSION}/llvm-project-${VERSION}.src.tar.xz
+wget https://github.com/llvm/llvm-project/releases/download/llvmorg-${VERSION}/llvm-project-${VERSION}.src.tar.xz.sig
+
+# Hard code signature key id?
+gpg --recv-keys --keyserver hkps://keyserver.ubuntu.com 474E22316ABF4785A88C6E8EA2C794A986419D8A
```

Comment:
Is this hard-coding likely to cause issues, and if so under what circumstances?

Reply:
I think hard-coding will only cause issues if a new version comes out that is signed with a different key or if the tar file is compromised. So, perhaps, hard coding it is acceptable. However, again, if we allow the user to choose different LLVM versions, then it is possible that some LLVM versions are signed with one key and other versions are signed with a different key. So, there are some issues if we want to be as general as possible, but maybe it is still acceptable? If we wanted to be as general as possible we can have a list of key ids hardcoded that we know and trust and share that between all Docker images / build processes.

Reply:
Sounds good to me. I think we can keep this for now then, and look at updating as needed.